modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java

```diff
@@ -104,184 +122,144 @@ public void setDownloadurl(String downloadurl) {
-                    break;
-                default:
-                    if (key.startsWith(SW360Attributes.RELEASE_EXTERNAL_ID_HASHES)) {
-                        if(hashes == null) {
-                            hashes = Collections.singleton(value);
-                        } else {
-                            hashes.add(value);
-                        }
-                    } else if (SW360CoordinateKeysToArtifactCoordinates.getValues().contains(key)) {
-                        if(coordinates == null) {
-                            coordinates = new HashMap<>();
-                            coordinates.put(key, value);
-                        } else {
-                            coordinates.put(key, value);
-                        }
-                    }
-            }
-        });
-    }
-
-    public SW360Release mergeWith(SW360Release releaseWithPrecedence) {
-        cpeid = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getCpeid);
-        downloadurl = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getDownloadurl);
-        mainLicenseIds = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getMainLicenseIds);
-        coordinates = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getCoordinates);
-        finalLicense = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getFinalLicense);
-        declaredLicense = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getDeclaredLicense);
-        observedLicense = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getObservedLicense);
-        releaseTagUrl = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getReleaseTagUrl);
-        softwareHeritageId = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getSoftwareHeritageId);
-        hashes = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getHashes);
-        clearingState = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getClearingState);
-        changeStatus = getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getChangeStatus);
-        copyrights =  getDominantGetterFromVariableMergeOrNull(releaseWithPrecedence, SW360Release::getCopyrights);
-
-        return this;
+    @Override
+    public SW360ReleaseLinkObjects createEmptyLinks() {
+        return new SW360ReleaseLinkObjects();
     }
 
-    private <T> T getDominantGetterFromVariableMergeOrNull(SW360Release release, Function<SW360Release, T> getter) {
-        return Optional.ofNullable(getter.apply(release))
-                .orElse(getter.apply(this));
+    @Override
+    public SW360ReleaseEmbedded createEmptyEmbedded() {
+        return new SW360ReleaseEmbedded();
     }
 
     @Override
```

Comment:
This equal method will always be false for a release gotten from an sw360 instance vs one created via antenna, since the `super.equals(o)` will always be false, since the antenna created sw360 release `_links` and `_embedded` are always null. 

This will lead to the case, that when updating is activated the sw360 release response will never equal the internally created release.

Reply:
I think this should already be fixed

Reply:
Additionally:
ExternalIds equals also is false if the map entries are not in the same order. 
so: 
Map1: declared_license=Apache-2.0, mvn=junit:junit:4.12
Map2: mvn=junit:junit:4.12, declared_license=Apache-2.0
would not be equal.

Reply:
the equals method on `Map` should ignore ordering. So that should not be a problem